rfcs/20240308-md-serialization/README.md

```diff
@@ -0,0 +1,135 @@
+# Proposal for memory descriptor serialization
+
+Compilers like Torch Inductor and XLA are gaining popularity and
+adding capabilities like Ahead of Time (AOT) compilation.  With those
+additions, it puts additional requirements to oneDNN to support
+serialization of objects.
+
+In particular, these compiler take as input a computational graph, and
+generate an executable binary from it.
```

Comment:
Not directly relate to this RFC, but curious to know if the input shapes will change after the compilation and weight pre-packing.

Reply:
This is independent of persistent cache I guess. As soon as user rely on pre-pack weights to reuse with primitive objects that were created differently (either with different params, or different environment), it is the user responsibility to check that memory descriptors match, and reorder if they don't.

But this is a fair point and this will be stressed out in documentation.

Reply:
Yes, absolutely it will be user responsibility to check that memory descriptors match. But I hope that they will not rely on querying memory descriptor from primitive descriptor and then comparing with the one recovered from deserialization during runtime. Otherwise, it will defeat the advantages of this proposal.

Reply:
They will have to, and that is fine IMHO.
When serialization is used properly: both md and primitive are serialized,  and deserialization environment is consistent with serialization environment, those two will match and all is fine.
If serialization was used improperly (e.g. change in environment, or serialize for one primitive configuration and deserialize with another), then things could mismatch, and the check/reorder should be there to prevent any issue.